### PR TITLE
points onboarding animations

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)

- correct animations
- now sending users to but and swap at the end of registration, but if there's no balance and swap if there is
- blocking watched wallets

## Screen recordings / screenshots

https://github.com/rainbow-me/browser-extension/assets/12115171/ed52ac03-dfe5-4e0e-b28f-9b7195aa9c81

https://github.com/rainbow-me/browser-extension/assets/12115171/acac8043-e42a-42a2-8508-f9e3d716ec28

### CODE REVIEW

File: src/entries/popup/pages/home/Points/utils.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        total_collections: 0,
+        usd_amount: 1,
+      },
+      type: 'rainbow-swaps' as CATEGORY_TYPE,
+      display_type: 'USD_AMOUNT' as CATEGORY_DISPLAY_TYPE,
+      earnings: { total: 0 },
+    },
+  ],
+  earnings: { total: 100 },
+};
```
Comment (greg-schrammel) [inline src/entries/popup/pages/home/Points/utils.ts:106]: is this used?

**Review by greg-schrammel — APPROVED** [review]
I think some colors could change, don't like that green lol, but the animations are cool!

File: src/design-system/components/AnimatedText/AnimatedText.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  customTypingSpeed,
+  ...textProps
+}: TextProps & {
+  id?: string;
+  delay?: number;
+  direction?: AnimationDirection;
+  customTypingSpeed?: number;
+}) => {
+  if (typeof children !== 'string') {
+    console.error('AnimatedText expects a string as children');
```
Comment (brunobar79) [inline src/design-system/components/AnimatedText/AnimatedText.tsx:24]: do we wanna leave this here?

File: src/entries/popup/pages/home/Points/utils.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import { i18n } from '~/core/languages';
+
+import {
+  CATEGORY_DISPLAY_TYPE,
+  CATEGORY_TYPE,
+  EXISTING_USER_ERROR,
+  INVALID_REFERRAL_CODE_ERROR,
+} from './references';
+
+export const CHARACTER_TYPING_SPEED = 0.1;
```
Comment (brunobar79) [inline src/entries/popup/pages/home/Points/utils.ts:10]: IMO this is a bit too slow. I'd lower it down to 0.05 or check what the app is using cause it def. seems faster there.

**Review by brunobar79 — APPROVED** [review]
💯

File: src/entries/popup/pages/home/Points/utils.ts (hunk 1 above)
Comment (estebanmino) [reply src/entries/popup/pages/home/Points/utils.ts:106]: removing

**Comment by estebanmino** [comment]
> I think some colors could change, don't like that green lol, but the animations are cool!

@greg-schrammel yeah, i changed the color for the one in figma

File: src/entries/popup/pages/home/Points/utils.ts (hunk 3 above)
Comment (estebanmino) [reply src/entries/popup/pages/home/Points/utils.ts:10]: i followed the app that had 0.1

File: src/design-system/components/AnimatedText/AnimatedText.tsx (hunk 2 above)
Comment (estebanmino) [reply src/design-system/components/AnimatedText/AnimatedText.tsx:24]: removing, was copy pasta

Comment (estebanmino) [reply src/design-system/components/AnimatedText/AnimatedText.tsx:24]: actually i need it, i put `children.split('')` just below to animate the characters

**Comment by estebanmino** [comment]
also, updated action buttons

image
image

File: src/entries/popup/pages/home/Points/utils.ts (hunk 3 above)
Comment (estebanmino) [reply src/entries/popup/pages/home/Points/utils.ts:5]: oh i just found 20 in other place, switching

**Pull request merged**